latch_cli/docker_utils/__init__.py

```diff
@@ -0,0 +1,233 @@
+                commands=[
+                    dedent(
+                        r"""
+                        run apt-get update --yes && \
+                            apt-get install --yes software-properties-common && \
+                            add-apt-repository "deb http://cloud.r-project.org/bin/linux/debian buster-cran40/" && \
+                            apt-get install --yes r-base r-base-dev libxml2-dev libcurl4-openssl-dev libssl-dev wget
+                        """
+                    ).strip(),
+                    "copy environment.R /opt/latch/environment.R",
+                    "run Rscript /opt/latch/environment.R",
+                ],
+                order=DockerCmdBlockOrder.precopy,
+            )
+        )
+
+    conda_env_file_name = None
+    if (pkg_root / "environment.yml").exists():
+        conda_env_file_name = "environment.yml"
+    elif (pkg_root / "environment.yaml").exists():
+        conda_env_file_name = "environment.yaml"
+
+    if conda_env_file_name is not None:
+        print("Adding conda + conda environment installation phase")
+        with (pkg_root / conda_env_file_name).open("rb") as f:
+            conda_env = yaml.safe_load(f)
+
+        if "name" in conda_env:
+            env_name = conda_env["name"]
+        else:
+            env_name = "workflow"
+
+        commands += [
+            DockerCmdBlock(
+                comment="set conda environment variables",
+                commands=[
+                    "env CONDA_DIR /opt/conda",
+                    "env PATH=$CONDA_DIR/bin:$PATH",
+                ],
+                order=DockerCmdBlockOrder.precopy,
+            ),
+            DockerCmdBlock(
+                comment="install miniconda",
+                commands=[
+                    dedent(
+                        r"""
+                        run apt-get update --yes && \
+                            apt-get install --yes curl && \
+                            curl -O https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh && \
+                            mkdir /root/.conda && \
+                            # docs for -b and -p flags: https://docs.anaconda.com/anaconda/install/silent-mode/#linux-macos
+                            bash Miniconda3-latest-Linux-x86_64.sh -b -p /opt/conda && \
+                            rm -f Miniconda3-latest-Linux-x86_64.sh && \
+                            conda init bash
+                        """
+                    ).strip(),
+                ],
+                order=DockerCmdBlockOrder.precopy,
+            ),
+            DockerCmdBlock(
+                comment="build and configure conda environment",
+                commands=[
+                    f"copy {conda_env_file_name} /opt/latch/{conda_env_file_name}",
+                    f"run conda env create --file /opt/latch/{conda_env_file_name} --name {env_name}",
+                    f"""shell ["conda", "run", "--name", "{env_name}", "/bin/bash", "-c"]""",
```

Comment:
```suggestion
                    f'shell ["conda", "run", "--name", "{env_name}", "/bin/bash", "-c"]',
```
another nit but no need for a multiline string here